crates/bevy_reflect/bevy_reflect_derive/src/lib.rs

```diff
@@ -94,11 +132,17 @@ pub fn reflect_trait(args: TokenStream, input: TokenStream) -> TokenStream {
 
 #[proc_macro]
 pub fn impl_reflect_value(input: TokenStream) -> TokenStream {
-    let def = parse_macro_input!(input as ReflectValueDef);
+    let def = parse_macro_input!(input as NamedReflectValueDef);
+
+    let reflected_type_name = def
+        .reflected_type_name
+        .unwrap_or_else(|| def.def.type_name.to_string());
```

Comment:
Nit: This could probably be encapsulated in a method. Then just call `def. reflected_type_name()`. This would help cleanup this function and separate concerns a bit more (and just be more modular in case we need to do this anywhere else).